src/Services/Exporter/ExporterWishlistToPdf.php

```diff
@@ -0,0 +1,117 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Twig\Environment;
+
+final class ExporterWishlistToPdf implements ExporterWishlistToPdfInterface
+{
+    private bool $isSelected = false;
+
+    private ProductVariantRepositoryInterface $productVariantRepository;
+
+    private VariantImagePathResolverInterface $variantImagePathResolver;
+
+    private VariantPdfModelFactoryInterface $variantPdfModelFactory;
+
+    private Environment $twigEnvironment;
+
+    private FlashBagInterface $flashBag;
+
+    private TranslatorInterface $translator;
+
+
+    public function __construct(
+        ProductVariantRepositoryInterface $productVariantRepository,
+        VariantImagePathResolverInterface $variantImagePathResolver,
+        VariantPdfModelFactoryInterface $variantPdfModelFactory,
+        Environment $twigEnvironment,
+        FlashBagInterface $flashBag,
+        TranslatorInterface $translator
+    ) {
+        $this->productVariantRepository = $productVariantRepository;
+        $this->variantImagePathResolver = $variantImagePathResolver;
+        $this->variantPdfModelFactory = $variantPdfModelFactory;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->flashBag = $flashBag;
+        $this->translator = $translator;
+    }
+
+    public function createModelToPdfAndExportToPdf(Collection $wishlistProducts, Request $request): void
+    {
+        $selectedProducts = [];
+
+        /** @var WishlistItemInterface $wishlistProduct */
+        foreach ($wishlistProducts as $wishlistProduct) {
+            if ($wishlistProduct->isSelected()) {
+                $this->isSelected = true;
+
+                $variant = $this->productVariantRepository->find($wishlistProduct->getWishlistProduct()->getVariant());
+
+                if (null === $variant || null === $wishlistProduct) {
+                    throw new ProductVariantNotFoundException(
+                        sprintf('The Product does not exist')
+                    );
+                }
+
+                $cartItem = $wishlistProduct->getCartItem()->getCartItem();
+                $quantity = $cartItem->getQuantity();
+                $baseUrl = $request->getSchemeAndHttpHost();
+                $urlToImage = $this->variantImagePathResolver->resolve($variant, $baseUrl);
+                $actualVariant = $cartItem->getVariant()->getCode();
+                $selectedProducts[] = $this->variantPdfModelFactory->createWithVariantAndImagePath(
+                    $variant,
+                    $urlToImage,
+                    $quantity,
+                    $actualVariant
+                );
+            }
+        }
+        $this->exportToPdf($selectedProducts);
+    }
+
+    private function exportToPdf(array $selectedProducts): void
+    {
+        if ( $this->isSelected === false) {
+            $this->flashBag->add(
```

Comment:
Is this a concern of exporter class or scheduling controller? Why not throw an exception here an handle it in controller?